### PR TITLE
BUG: Period.__eq__ numpy scalar (#44182 (comment))

### PR DESCRIPTION
**Pull request by mathause**
- [x] fixes https://github.com/pandas-dev/pandas/pull/44182#issuecomment-956851322
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

**Comment by mathause** [comment]
I added three tests but only one of them fails without the code change.

Fixed by the code update:
```python
Period("2000-01", "M") == np.array(0)
```

Additional tests not affected by the code update:
```python
idx = period_range("2007-01", periods=20, freq="M")
idx == np.array(idx[10])

Period("2000-01", "M") == np.int_(0)
```

File: pandas/_libs/tslibs/period.pyx [hunk 1]
```diff
@@ -1656,7 +1656,7 @@ cdef class _Period(PeriodMixin):
             return PyObject_RichCompareBool(self.ordinal, other.ordinal, op)
         elif other is NaT:
             return _nat_scalar_rules[op]
-        elif util.is_array(other):
+        elif util.is_array(other) and not np.ndim(other) == 0:
```
Comment (jreback) [inline pandas/_libs/tslibs/period.pyx:1659]: np.ndim(..) > 0

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by mathause** [comment]
This used to return `True`, so I think we are not yet entirely there:

```python
p = pd.Period("2000-01", "M")
p == np.array(p)
```

File: pandas/_libs/tslibs/period.pyx [hunk 2]
```diff
@@ -1657,8 +1657,11 @@ cdef class _Period(PeriodMixin):
         elif other is NaT:
             return _nat_scalar_rules[op]
         elif util.is_array(other):
-            # in particular ndarray[object]; see test_pi_cmp_period
-            return np.array([PyObject_RichCompare(self, x, op) for x in other])
+            if np.ndim(other) > 0:
```
Comment (jbrockmendel) [inline pandas/_libs/tslibs/period.pyx:1660]: cnp.PyArray_IsZeroDim(other)

Comment (jbrockmendel) [reply pandas/_libs/tslibs/period.pyx:1660]: `# GH#44285`

File: pandas/tests/scalar/period/test_period.py [hunk 3]
```diff
@@ -1148,6 +1148,16 @@ def test_period_cmp_nat(self):
...
             assert not left >= right
 
+    @pytest.mark.parametrize(
+        "scalar, expected",
+        ((np.array(0), False), (np.int_(0), False), (Period("2000-01", "M"), True)),
+    )
+    def test_comparison_numpy_scalar(self, scalar, expected):
+        p = Period("2000-01", "M")
+
+        for left, right in ((p, scalar), (scalar, p)):
```
Comment (jbrockmendel) [inline pandas/tests/scalar/period/test_period.py:1158]: the loop here is unnecessary; its two lines either way and clearer without

File: pandas/tests/scalar/period/test_period.py [hunk 4]
```diff
@@ -1148,6 +1148,16 @@ def test_period_cmp_nat(self):
             assert not left <= right
             assert not left >= right
 
+    @pytest.mark.parametrize(
+        "scalar, expected",
+        ((np.array(0), False), (np.int_(0), False), (Period("2000-01", "M"), True)),
+    )
+    def test_comparison_numpy_scalar(self, scalar, expected):
```
Comment (jbrockmendel) [inline pandas/tests/scalar/period/test_period.py:1155]: scalars are already tested; its zerodim we need to catch.  in particular a zero-dim ndarray containing a Period object

**Review by jbrockmendel — APPROVED** [review]
LGTM cc @jreback

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @mathause

**Pull request merged**